extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/VertxVaultClient.java

```diff
@@ -377,143 +386,147 @@ public void deletePolicy(String token, String name) {
+        return exec(request, body, null, expectedCode);
     }
 
     protected <T> T put(String path, String token, Object body, Class<T> resultClass) {
-        Request request = builder(path, token).put(requestBody(body)).build();
-        return exec(request, resultClass);
+        HttpRequest<Buffer> request = builder(path, token).method(HttpMethod.PUT);
+        return exec(request, body, resultClass);
     }
 
     protected <T> T put(String path, Object body, Class<T> resultClass) {
-        Request request = builder(path).put(requestBody(body)).build();
-        return exec(request, resultClass);
+        HttpRequest<Buffer> request = builder(path).method(HttpMethod.PUT);
+        return exec(request, body, resultClass);
     }
 
     protected <T> T get(String path, String token, Class<T> resultClass) {
-        Request request = builder(path, token).get().build();
+        HttpRequest<Buffer> request = builder(path, token).method(HttpMethod.GET);
         return exec(request, resultClass);
     }
 
     protected <T> T get(String path, Map<String, String> queryParams, Class<T> resultClass) {
-        final Request request = builder(path, queryParams).get().build();
+        final HttpRequest<Buffer> request = builder(path, queryParams).method(HttpMethod.GET);
         return exec(request, resultClass);
     }
 
     protected int head(String path) {
-        final Request request = builder(path).head().build();
+        final HttpRequest<Buffer> request = builder(path).method(HttpMethod.HEAD);
         return exec(request);
     }
 
     protected int head(String path, Map<String, String> queryParams) {
-        final Request request = builder(path, queryParams).head().build();
+        final HttpRequest<Buffer> request = builder(path, queryParams).method(HttpMethod.HEAD);
         return exec(request);
     }
 
-    private <T> T exec(Request request, Class<T> resultClass) {
-        return exec(request, resultClass, 200);
+    private <T> T exec(HttpRequest<Buffer> request, Class<T> resultClass) {
+        return exec(request, null, resultClass, 200);
     }
 
-    private <T> T exec(Request request, int expectedCode) {
-        return exec(request, null, expectedCode);
+    private <T> T exec(HttpRequest<Buffer> request, int expectedCode) {
+        return exec(request, null, null, expectedCode);
     }
 
-    private <T> T exec(Request request, Class<T> resultClass, int expectedCode) {
-        try (Response response = client.newCall(request).execute()) {
-            if (response.code() != expectedCode) {
+    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass) {
+        return exec(request, body, resultClass, 200);
+    }
+
+    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass, int expectedCode) {
+        try {
+            HttpResponse<Buffer> response = body == null
+                    ? request.sendAndAwait()
```

Comment:
We may want to be a bit more careful here. 
This may wait.... forever.

I would do:
```
request.send()
  .await().atMost(Duration.ofSecond(a_sensible_and_configurable_amount);
```

Reply:
Note that await is forbidden when you are on the event loop / I/O thread

Reply:
I thought we would be protected with the idle timeout:
```
        WebClientOptions options = new WebClientOptions()
                .setConnectTimeout((int) vaultBootstrapConfig.connectTimeout.getSeconds())
                .setIdleTimeout((int) vaultBootstrapConfig.readTimeout.getSeconds());
```

is this equivalent to a read timeout? I looked at the definition in the doc, and it seems different. what is the proper way to do a request timeout? should that be on the request (using `HttpClientRequest.setTimeout`?), or something like this:

```
    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass, int expectedCode) {
        try {
            Uni<HttpResponse<Buffer>> uni = body == null 
                    ? request.send() 
                    : request.sendBuffer(Buffer.buffer(requestBody(body)));
            HttpResponse<Buffer> response = uni.await().atMost(vaultConfigHolder.getVaultBootstrapConfig().readTimeout);
```

Reply:
done. I reused the read timeout, which is part of the config.

Reply:
or should that be:
```
            HttpRequest<Buffer> httpRequest = request.timeout(getRequestTimeout().toMillis());
            Uni<HttpResponse<Buffer>> uni = body == null 
                    ? httpRequest.send()
                    : httpRequest.sendBuffer(Buffer.buffer(requestBody(body)));
            HttpResponse<Buffer> response = uni.await().indefinitely();
```
how can we timeout at the socket level?